wagtail/contrib/wagtailfrontendcache/backends.py

```diff
@@ -84,3 +88,52 @@ def purge(self, url):
         if response_json['result'] == 'error':
             logger.error("Couldn't purge '%s' from Cloudflare. Cloudflare error '%s'", url, response_json['msg'])
             return
+
+
+class CloudfrontBackend(BaseBackend):
+    def __init__(self, params):
+        self.client = boto3.client('cloudfront')
+        try:
+            self.cloudfront_distribution_id = params.pop('DISTRIBUTION_ID')
+        except KeyError:
+            raise ImproperlyConfigured(
+                "The setting 'WAGTAILFRONTENDCACHE' requires the object 'DISTRIBUTION_ID'."
+            )
+
+    def purge(self, url):
+        url_parsed = urlparse(url)
+        distribution_id = None
+
+        if isinstance(self.cloudfront_distribution_id, dict):
+            host = url_parsed.hostname
+            if host in self.cloudfront_distribution_id:
+                distribution_id = self.cloudfront_distribution_id.get(host)
+            else:
+                logger.error(
```

Comment:
It's possible that someone might want cloudfront on one site but not another, so perhaps this should be info rather than error?